os/mm/kmm_heap/kmm_memalign.c

```diff
@@ -144,16 +142,14 @@ FAR void *kmm_memalign(size_t alignment, size_t size)
 	if (size == 0) {
 		return NULL;
 	}
+
+	mmaddress_t caller_retaddr = NULL;	/* for generalising the call to mm_memalign api */
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	mmaddress_t caller_retaddr = GET_RETURN_ADDRESS();
+	caller_retaddr = GET_RETURN_ADDRESS();
 #endif
```

Comment:
Usually we define the config, then it will do like below
1. assign null to the variable
2. reassign new address to the variable again

To avoid unnecessary assignment for performance, how about below?
```
mmaddress_t caller_retaddr;   (at start of the function)

if (size == 0) {
  ...
}

#ifdef CONFIG_....
caller... = GET_...();
#else
caller = NULL;
#endif
```